[Merged by Bors] - Add cli option for voluntary exits (#1781)
sigp/lighthouse: book/src/voluntary-exit.md

# Voluntary exits

> A validator may chose to voluntary stop performing duties (proposing blocks and attesting to blocks) by submitting

Reviewer: ```suggestion
A validator may chose to voluntarily stop performing duties (proposing blocks and attesting to blocks) by submitting
```

a voluntary exit transaction to the beacon chain.

> A validator can initiate a voluntary exit provided that the validator is currently active (has not been slashed) and has been active for at least 256 epochs (~27 hours) since it has been activated.

Reviewer: ```suggestion
A validator can initiate a voluntary exit provided that the validator is currently active, has not been slashed and has been active for at least 256 epochs (~27 hours) since it has been activated.
```

> Note: After initiating a voluntary exit, the validator will have to keep performing duties until it has successfully exited to avoid penalties.

> It takes at minimum 5 epochs (32 minutes) for a validator to exit after initiating a voluntary exit.

Reviewer: ```suggestion
It takes at a minimum 5 epochs (32 minutes) for a validator to exit after initiating a voluntary exit.
```

> This number can be much higher depending on how many other validators are in queue to exit.

Reviewer: ```suggestion
This number can be much higher depending on how many other validators are queued to exit.
```

## Withdrawal of exited funds

> Even though users can perform a voluntary exit in phase 0, they **cannot withdraw their exited funds until phase 2**.

Reviewer: ```suggestion
Even though users can perform a voluntary exit in phase 0, they **cannot withdraw their exited funds at this point in time**.
```

Reviewer: To avoid committing to a withdrawal-is-possible date.

This implies that the staked funds are effectively **frozen** until withdrawals are enabled in future phases.

To understand the phased rollout strategy for Eth2, please visit <https://ethereum.org/en/eth2/#roadmap>.

## Initiating a voluntary exit

> In order to initiate a exit, users can use the `lighthouse account validator exit` command.

Reviewer: ```suggestion
In order to initiate an exit, users can use the `lighthouse account validator exit` command.
```

> The `--validator` flag is used to specify the validator for which the user wants to initiate an exit transaction.

Reviewer: If we don't just use an EIP-2335 keystore path instead, I think we should specify how the value of this flag identifies a validator.

If we *do* just use an EIP-2335 keystore path, you can ignore this comment.

> The `--beacon-node` flag is used to specify a beacon chain http endpoint that confirms to the [standard api](https://ethereum.github.io/eth2.0-APIs/) specifications which will be used to validate and propagate the voluntary exit. The default value for this flag is `http://localhost:5052`.

Reviewer: ```suggestion
The `--beacon-node` flag is used to specify a beacon chain HTTP endpoint that confirms to the [Eth2.0 Standard API](https://ethereum.github.io/eth2.0-APIs/) specifications. That beacon node will be used to validate and propagate the voluntary exit. The default value for this flag is `http://localhost:5052`.
```

The `--testnet` flag is used to specify a particular testnet (default is `medalla`).

> The `exit` command will search for the provided validator in the `validators-dir` (default `~/.lighthouse/{testnet}/validators`) directory and validator password file in `secrets-dir` (default `  ~/.lighthouse/{testnet}/secrets`).

Reviewer: ```suggestion
The `exit` command will search for the provided validator in the `validators-dir` (default `~/.lighthouse/{testnet}/validators`) directory and for the validator's password file in `secrets-dir` (default `  ~/.lighthouse/{testnet}/secrets`).
```

> If the corresponding  password file is not found in the `secrets-dir`, the user will be prompted for a password on the terminal.

Reviewer: ```suggestion
If the corresponding password file is not found in the `secrets-dir`, the user will be prompted for a password on the terminal.
```

> [//]: # (Word this better or remove datadir part entirely)

Reviewer: TODO
